pr_agent/tools/pr_similar_issue.py

```diff
@@ -19,19 +19,15 @@
 
 class PRSimilarIssue:
     def __init__(self, issue_url: str, args: list = None):
-        if get_settings().config.git_provider != "github":
-            raise Exception("Only github is supported for similar issue tool")
 
         self.cli_mode = get_settings().CONFIG.CLI_MODE
         self.max_issues_to_scan = get_settings().pr_similar_issue.max_issues_to_scan
         self.issue_url = issue_url
         self.git_provider = get_git_provider()()
-        repo_name, issue_number = self.git_provider._parse_issue_url(issue_url.split('=')[-1])
-        self.git_provider.repo = repo_name
-        self.git_provider.repo_obj = self.git_provider.github_client.get_repo(repo_name)
+        self.git_provider.repo_obj  = self.git_provider.parse_issue_url_and_get_repo_obj(issue_url.split('=')[-1])
```

Comment:
**Suggestion:** Consider using a helper function to parse the issue URL, as this operation is repeated multiple times in the code. This will make the code more DRY (Don't Repeat Yourself) and easier to maintain. [maintainability]